### PR TITLE
RM#51430 ACCOUNT CONFIG : paymentVouchersOnInvoice

### PR DESCRIPTION
**Pull request by hmi-axelor**

### CODE REVIEW

File: axelor-account/src/main/java/com/axelor/apps/account/service/InvoiceVisibilityServiceImpl.java [hunk 1]
```diff
@@ -189,4 +196,12 @@ protected boolean _getInvoiceTermsCondition2(Invoice invoice) {
   protected boolean _getDecisionDateCondition(Invoice invoice) {
     return invoice.getDecisionPfpTakenDate() != null;
   }
+
+  protected boolean _getStatusPaymentVouchers(Invoice invoice) {
+    AppAccountService appAccount = Beans.get(AppAccountService.class);
+    return (invoice.getOperationTypeSelect() == InvoiceRepository.OPERATION_TYPE_CLIENT_SALE
+            || invoice.getOperationTypeSelect() == InvoiceRepository.OPERATION_TYPE_SUPPLIER_REFUND)
+        ? appAccount.getAppAccount().getPaymentVouchersOnCustomerInvoice()
+        : appAccount.getAppAccount().getPaymentVouchersOnSupplierInvoice();
```
Comment (gdu-axelor) [inline axelor-account/src/main/java/com/axelor/apps/account/service/InvoiceVisibilityServiceImpl.java:205]: Should be .OPERATION_TYPE_CLIENT_SALE || .OPERATION_TYPE_CLIENT_REFUND

Comment (gdu-axelor) [reply axelor-account/src/main/java/com/axelor/apps/account/service/InvoiceVisibilityServiceImpl.java:205]: We already have a method for that : 
https://github.com/axelor/axelor-open-suite/blob/master/axelor-account/src/main/java/com/axelor/apps/account/service/invoice/InvoiceToolService.java#L92

File: axelor-account/src/main/java/com/axelor/apps/account/service/payment/paymentvoucher/PaymentVoucherConfirmService.java [hunk 2]
```diff
@@ -165,7 +166,24 @@ public void confirmPaymentVoucher(PaymentVoucher paymentVoucher) throws AxelorEx
     // TODO RECUPERER DEVISE DE LA PREMIERE DETTE
     // Currency currencyToPay = null;
 
-    createMoveAndConfirm(paymentVoucher, journal, paymentModeAccount, isPaymentValueForCollection);
+    AppAccountService appAccountService = Beans.get(AppAccountService.class);
```
Comment (gdu-axelor) [inline axelor-account/src/main/java/com/axelor/apps/account/service/payment/paymentvoucher/PaymentVoucherConfirmService.java:169]: No need to take into account the conditions here

**Review by gdu-axelor — CHANGES_REQUESTED** [review]

File: axelor-account/src/main/java/com/axelor/apps/account/service/InvoiceVisibilityServiceImpl.java (hunk 1 above)
Comment (rme-axelor) [reply axelor-account/src/main/java/com/axelor/apps/account/service/InvoiceVisibilityServiceImpl.java]: But in ticket it's different.

It's
Supplier Invoice or customer refund
Customer Invoice or supplier refund

**Pull request merged**